src/widget/wtracktableview.cpp

```diff
@@ -1606,6 +1612,25 @@ void WTrackTableView::slotClearBeats() {
     }
 }
 
+void WTrackTableView::slotClearWaveform() {
+    TrackModel* trackModel = getTrackModel();
+    if (trackModel == nullptr) {
+        return;
+    }
+
+    AnalysisDao& analysisDao = m_pTrackCollection->getAnalysisDAO();
+    QModelIndexList indices = selectionModel()->selectedRows();
+    for (const QModelIndex& index : indices) {
+        TrackPointer pTrack = trackModel->getTrack(index);
+        if (!pTrack) {
+            continue;
+        }
+        analysisDao.deleteAnalysesForTrack(pTrack->getId());
```

Comment:
Will this delete the all analysis data? 
I think we should rethink the use-cases for "ClearWaveform" and name this feature accordingly. 
IMHO clearing all analysis makes sense if the use-case is, that the track samples have changed. 
In this case all position related data is invalid.
What will happen with cue points?